File: Java/IsBalanced.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int getHeight(TreeNode root) {
        if(root == null) return 0;

        int left = getHeight(root.left);
        int right = getHeight(root.right);

        return Math.max(left, right) + 1;
    }

    public boolean isBalanced(TreeNode root) {
        if(root == null) return true;

        return Math.abs(getHeight(root.left) - getHeight(root.right)) <= 1 &&
                isBalanced(root.left) && isBalanced(root.right);
    }
}

File: Go/is_balanced.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func getHeight(root *TreeNode) int {
    if root == nil {
        return 0
    }
    left := getHeight(root.Left)
    right := getHeight(root.Right)

    if left > right {
        return left + 1
    }
    return right + 1
}

func abs(n int) int {
    if n < 0 {
        return -n;
    }
    return n;
}

func isBalanced(root *TreeNode) bool {
    if root == nil {
        return true
    }

    return abs(getHeight(root.Left) - getHeight(root.Right)) <= 1 &&
            isBalanced(root.Left) && isBalanced(root.Right)
}